src/openrct2-ui/windows/LoadSave.cpp

```diff
@@ -427,694 +471,639 @@ static u8string Browse(bool isSave)
             }
-            else
+            // display a marker next to the currently loaded game file
+            if (_listItems[i].loaded)
             {
-                context_show_error(STR_SAVE_LANDSCAPE, STR_LANDSCAPE_SAVE_FAILED, {});
-                WindowLoadsaveInvokeCallback(MODAL_RESULT_FAIL, pathBuffer);
+                auto ft = Formatter();
+                ft.Add<StringId>(STR_RIGHTGUILLEMET);
+                DrawTextBasic(&dpi, { 0, y }, stringId, ft);
             }
-            break;
 
-        case (LOADSAVETYPE_SAVE | LOADSAVETYPE_SCENARIO):
-        {
-            SetAndSaveConfigPath(gConfigGeneral.LastSaveScenarioDirectory, pathBuffer);
-            int32_t parkFlagsBackup = gParkFlags;
-            gParkFlags &= ~PARK_FLAGS_SPRITES_INITIALISED;
-            gEditorStep = EditorStep::Invalid;
-            gScenarioFileName = std::string(String::ToStringView(pathBuffer, std::size(pathBuffer)));
-            int32_t success = scenario_save(pathBuffer, gConfigGeneral.SavePluginData ? 3 : 2);
-            gParkFlags = parkFlagsBackup;
+            // Print filename
+            auto ft = Formatter();
+            ft.Add<StringId>(STR_STRING);
+            ft.Add<char*>(_listItems[i].name.c_str());
+            int32_t max_file_width = widgets[WIDX_SORT_NAME].width() - 10;
+            DrawTextEllipsised(&dpi, { 10, y }, max_file_width, stringId, ft);
 
-            if (success)
-            {
-                window_close_by_class(WindowClass::Loadsave);
-                WindowLoadsaveInvokeCallback(MODAL_RESULT_OK, pathBuffer);
-                title_load();
-            }
-            else
+            // Print formatted modified date, if this is a file
+            if (_listItems[i].type == TYPE_FILE)
             {
-                context_show_error(STR_FILE_DIALOG_TITLE_SAVE_SCENARIO, STR_SCENARIO_SAVE_FAILED, {});
-                gEditorStep = EditorStep::ObjectiveSelection;
-                WindowLoadsaveInvokeCallback(MODAL_RESULT_FAIL, pathBuffer);
+                ft = Formatter();
+                ft.Add<StringId>(STR_STRING);
+                ft.Add<char*>(_listItems[i].date_formatted.c_str());
+                DrawTextEllipsised(
+                    &dpi, { dateAnchor - DATE_TIME_GAP, y }, maxDateWidth, stringId, ft, { TextAlignment::RIGHT });
+
+                ft = Formatter();
+                ft.Add<StringId>(STR_STRING);
+                ft.Add<char*>(_listItems[i].time_formatted.c_str());
+                DrawTextEllipsised(&dpi, { dateAnchor + DATE_TIME_GAP, y }, maxTimeWidth, stringId, ft);
             }
-            break;
         }
+    }
+#pragma endregion
+};
 
-        case (LOADSAVETYPE_LOAD | LOADSAVETYPE_TRACK):
+rct_window* WindowLoadsaveOpen(
+    int32_t type, std::string_view defaultPath, std::function<void(int32_t result, std::string_view)> callback,
+    TrackDesign* trackDesign)
+{
+    _loadSaveCallback = callback;
+    _trackDesign = trackDesign;
+    _type = type;
```

Comment:
Sounds like `_type` can be a member variable, can't it? Seems like it is only used outside of the class a few lines below here, so it could be sent as a parameter of `OpenSystemFileBrowser`

Reply:
It's also used in the shared `Select` function, which is used outside of the class. This depends on the other window, and must be split clean it up - something I'd like to leave for another PR.